### PR TITLE
Feature careermode

### PR DESCRIPTION
**Pull request by jeroenoverman**
Added careermode to the game

-A player starts the game by entering its name. The game checks if his profile is available.

-Al player can choose which level he wants to play, but can only play one level higher than the level he has cleared

- a player can go to the shop after each level.

-profile data is written to a json file.

closes #109  #72 #66 #42

### CODE REVIEW

File: core/src/com/group66/game/helpers/LevelLoader.java [hunk 1]
```diff
@@ -13,9 +14,10 @@
     
     /**
      * Load a test level.
+     * @param level TODO
```
Comment (iiKoe) [inline core/src/com/group66/game/helpers/LevelLoader.java]: Needs more params

File: core/src/com/group66/game/helpers/ProfileManager.java [hunk 2]
```diff
@@ -0,0 +1,89 @@
...
+import org.json.JSONObject;
+
+import com.badlogic.gdx.Gdx;
+import com.badlogic.gdx.files.FileHandle;
+import com.group66.game.BustaMove;
+import com.group66.game.logging.MessageType;
+import com.group66.game.settings.DynamicSettings;
+
+/**
+ * @author Jeroen
```
Comment (iiKoe) [inline core/src/com/group66/game/helpers/ProfileManager.java]: No one else does this, can you remove?

File: core/src/com/group66/game/input/LevelSelectInputListener.java [hunk 3]
```diff
@@ -0,0 +1,43 @@
+/**
+ * 
+ */
+package com.group66.game.input;
+
+import com.badlogic.gdx.Input.TextInputListener;
+import com.group66.game.settings.DynamicSettings;
+
+/**
+ * @author Jeroen
```
Comment (iiKoe) [inline core/src/com/group66/game/input/LevelSelectInputListener.java]: Again

File: core/src/com/group66/game/input/NameInputListener.java [hunk 4]
```diff
@@ -0,0 +1,40 @@
+/**
+ * 
+ */
+package com.group66.game.input;
+
+import com.badlogic.gdx.Input.TextInputListener;
+import com.group66.game.settings.DynamicSettings;
+
+/**
+ * @author Jeroen
```
Comment (iiKoe) [inline core/src/com/group66/game/input/NameInputListener.java]: Again :p

File: core/src/com/group66/game/screens/CareerScreen.java [hunk 5]
```diff
@@ -0,0 +1,260 @@
...
+     * @param dynamicSettings
+     */
+    public CareerScreen(DynamicSettings dynamicSettings) {
+        this();
+    }
+
+    private void createScreen() {
+        stage = new Stage();
+        Gdx.input.setInputProcessor(stage);
+
```
Comment (iiKoe) [inline core/src/com/group66/game/screens/CareerScreen.java]: Extreme amount of newlines

File: core/src/com/group66/game/screens/OnePlayerGameScreen.java [hunk 6]
```diff
@@ -75,44 +80,56 @@ public void render(float delta) {
...
-        
+
         /* Check if game-over condition is reached */
         if (ballManager.isGameOver()) {
             BustaMove.getGameInstance().log(MessageType.Info, "Failed the level");
-            DynamicSettings ds = ballManager.getDynamicSettings();
-            if (ds.hasExtraLife()) {
-                ds.setExtraLife(false);
-                BustaMove.getGameInstance().log(MessageType.Info, "Keeping Dynamic Settings");
+            //DynamicSettings ds = ballManager.getDynamicSettings();
```
Comment (iiKoe) [inline core/src/com/group66/game/screens/OnePlayerGameScreen.java]: Remove commented code

File: core/src/com/group66/game/screens/ShopScreen.java [hunk 7]
```diff
@@ -149,28 +154,50 @@ private void createScreen() {
         // revert the checked state.
         levelButton.addListener(new ChangeListener() {
             public void changed(ChangeEvent event, Actor actor) {
-                BustaMove.getGameInstance().setScreen(new MainMenuScreen());
+                dispose();
+                try {
```
Comment (iiKoe) [inline core/src/com/group66/game/screens/ShopScreen.java]: If we don't do special stuff for different exceptions, we might as well do
`catch (Exception e)`

File: core/src/com/group66/game/screens/ShopScreen.java [hunk 8]
```diff
@@ -259,7 +286,7 @@ public void render(float delta) {
     @Override
     public void show() {
         // TODO Auto-generated method stub
```
Comment (iiKoe) [inline core/src/com/group66/game/screens/ShopScreen.java]: Remove todo's here

File: core/src/com/group66/game/screens/StartScreen.java [hunk 9]
```diff
@@ -101,14 +134,14 @@ public void changed(ChangeEvent event, Actor actor) {
...
         Gdx.gl.glClearColor(0.2f, 0.2f, 0.3f, 1);
         Gdx.gl.glClear(GL20.GL_COLOR_BUFFER_BIT);
-        
+
         /* Draw the background */
-        BustaMove.getGameInstance().batch.begin();
-        BustaMove.getGameInstance().batch.enableBlending();
-        BustaMove.getGameInstance().batch.draw(AssetLoader.youwinbg, Config.SINGLE_PLAYER_OFFSET, 0, Config.LEVEL_WIDTH,
+        game.batch.begin();
+        game.batch.enableBlending();
```
Comment (iiKoe) [inline core/src/com/group66/game/screens/StartScreen.java]: Why did you go back to non singleton?

File: core/src/com/group66/game/screens/YouLoseScreenCareer.java [hunk 10]
```diff
@@ -1,48 +1,59 @@
...
+import com.group66.game.BustaMove;
 import com.group66.game.helpers.AssetLoader;
+import com.group66.game.helpers.TextDrawer;
 import com.group66.game.settings.Config;
-import com.group66.game.BustaMove;
+import com.group66.game.settings.DynamicSettings;
 
 /**
- * A Class for the MainMenuScreen of the game.
+ * @author Jeroen
```
Comment (iiKoe) [inline core/src/com/group66/game/screens/YouLoseScreenCareer.java]: Miep

File: core/src/com/group66/game/screens/YouLoseScreenCareer.java [hunk 11]
```diff
@@ -1,48 +1,59 @@
...
     /**
-     * Instantiates a new main menu screen.
+     * @param dynamicSettings
      */
-    public YouLoseScreen() {
-        AssetLoader.load();
-        createScreen();
+    public YouLoseScreenCareer(DynamicSettings dynamicSettings) {
+        super(dynamicSettings);
+        // TODO Auto-generated constructor stub
```
Comment (iiKoe) [inline core/src/com/group66/game/screens/YouLoseScreenCareer.java]: Remove todo

File: core/src/com/group66/game/settings/Config.java [hunk 12]
```diff
@@ -112,4 +112,6 @@
     
     /** The Constant CURRENCY_Y for the text drawing the the ShopScreen. */
     public static final int CURRENCY_Y = 600;
+    
+    public static final int NUMBER_OF_LEVELS = 10;
```
Comment (iiKoe) [inline core/src/com/group66/game/settings/Config.java]: Add comment

**Comment by jeroenoverman** [comment]
Fixed you comments @iiKoe

**Pull request merged**